File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/ObjC/FileService/FileService/ViewController.m

#import "ViewController.h"
#import "ImagePreviewViewController.h"
#import "FileObject.h"
#import <Backendless-Swift.h>

@interface ViewController() {
    BOOL isSimulator;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
#if TARGET_IPHONE_SIMULATOR
    isSimulator = YES;
#else
    isSimulator = NO;
#endif
}

- (void)takePhoto:(id)sender {
    if (isSimulator) {
        [self presentViewController:[UIAlertController alertControllerWithTitle:@"Warning!" message:@"Make sure to run the app on a device, the functionality is not available in a simulator" preferredStyle:UIAlertControllerStyleAlert] animated:YES completion:^{
            [NSTimer scheduledTimerWithTimeInterval:3 repeats:NO block:^(NSTimer *timer) {
                [self dismissViewControllerAnimated:YES completion:nil];
            }];
        }];
    }
    else {
        UIImagePickerController *picker = [UIImagePickerController new];
        picker.delegate = self;
        if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
            [picker setSourceType:UIImagePickerControllerSourceTypeCamera];
            picker.showsCameraControls = YES;
        }
        else {
            [picker setSourceType:UIImagePickerControllerSourceTypePhotoLibrary];
        }
        [self presentViewController:picker animated:YES completion:nil];
    }    
}

- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary *)info {
    [self dismissViewControllerAnimated:YES completion:^{
        ImagePreviewViewController *imagePreviewVC = [[self storyboard] instantiateViewControllerWithIdentifier:@"ImagePreviewViewController"];
        imagePreviewVC.mainImage = [info objectForKey:@"UIImagePickerControllerOriginalImage"];
        [self.navigationController pushViewController:imagePreviewVC animated:YES];
    }];
}

@end
          

File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/Swift/FileService/FileService/ViewController.swift

import UIKit
import Backendless

class ViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {    
    
    var isSimulator: Bool = false
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
#if (arch(i386) || arch(x86_64))
        isSimulator = true
#else
        isSimulator = false
#endif
    }
    
    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
        self.dismiss(animated: true, completion: {
            let imagePreviewVC = self.storyboard?.instantiateViewController(withIdentifier: "ImagePreviewViewController") as! ImagePreviewViewController
            imagePreviewVC.mainImage = info[.originalImage] as? UIImage
            self.navigationController?.pushViewController(imagePreviewVC, animated: true)
        })
    }
    
    @IBAction func takePhoto(_ sender: Any) {
        if isSimulator {
            self.present(UIAlertController.init(title: "Warning!", message: "Make sure to run the app on a device, the functionality is not available in a simulator", preferredStyle: .alert), animated: true, completion: {
                Timer.scheduledTimer(withTimeInterval: 3, repeats:false, block: {_ in
                    self.dismiss(animated: true, completion: nil)
                })
            })
        }
        else {
            let picker = UIImagePickerController()
            picker.delegate = self
            if (UIImagePickerController.isSourceTypeAvailable(.camera)) {
                picker.sourceType = .camera
                picker.showsCameraControls = true
            }
            else {
                picker.sourceType = .photoLibrary
            }
            self.present(picker, animated: true, completion: nil)
        }
    }
}

